### DISCUSSION TITLE
Setup issues

### DISCUSSION THREAD

**Discussion by ixalor** [Almamu/EVESharp#19] [Q&A]
Installed and thought I setup everything properly however getting a snag when running the node.  Cluster is up fine but I am getting the following:

7/21/2021 7:56 AM D CacheStorage: Loading cache data for charCreationInfo.lipsticks of type Rowset
7/21/2021 7:56 AM I main: Cache Initialized
7/21/2021 7:57 AM I ItemFactory: Preloaded 13155 static items
7/21/2021 7:57 AM E main: System.AggregateException: One or more errors occurred. (The given item (1000132) is not loaded by this ItemManager: )
 ---> Node.Inventory.Exceptions.ItemNotLoadedException: The given item (1000132) is not loaded by this ItemManager:
   at Node.Inventory.ItemFactory.GetItem(Int32 itemID) in X:\EVESharp\EVESharp-master\Server\Node\Inventory\ItemFactory.cs:line 247
   at Node.Inventory.ItemFactory.Load() in X:\EVESharp\EVESharp-master\Server\Node\Inventory\ItemFactory.cs:line 143
   at Node.Inventory.ItemFactory.Init() in X:\EVESharp\EVESharp-master\Server\Node\Inventory\ItemFactory.cs:line 134
   at Node.Program.<>c__DisplayClass0_0.<InitializeItemFactory>b__0() in X:\EVESharp\EVESharp-master\Server\Node\Program.cs:line 68
   at System.Threading.Tasks.Task.InnerInvoke()
   at System.Threading.Tasks.Task.<>c.<.cctor>b__277_0(Object obj)
   at System.Threading.ExecutionContext.RunFromThreadPoolDispatchLoop(Thread threadPoolThread, ExecutionContext executionContext, ContextCallback callback, Object state)
--- End of stack trace from previous location ---
   at System.Threading.ExecutionContext.RunFromThreadPoolDispatchLoop(Thread threadPoolThread, ExecutionContext executionContext, ContextCallback callback, Object state)
   at System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task& currentTaskSlot, Thread threadPoolThread)
--- End of stack trace from previous location ---
   at Node.Program.InitializeItemFactory(Channel logChannel, Container dependencies) in X:\EVESharp\EVESharp-master\Server\Node\Program.cs:line 65
   --- End of inner exception stack trace ---
   at System.Threading.Tasks.Task.WaitAllCore(Task[] tasks, Int32 millisecondsTimeout, CancellationToken cancellationToken)
   at Node.Program.Main(String[] argv) in X:\EVESharp\EVESharp-master\Server\Node\Program.cs:line 272
7/21/2021 7:57 AM F main: Node stopped...

I am not sure which SQL file loads this particular item or if something else is out of wack, but feel free to let me know I did something wrong.

**Comment by ixalor** [comment]
This [below]  is what you get when you don't actually PATCH the client with all the patches.  Direction unclear that I need every one of those files, but I at least got to the point where the client sees the cluster and if/when the node issue is resolved, it should work.

But if you are finding this message because your client is giving a protocol error for EveSharp:

Install the client as directed.  Download each patch and run it one at a time, in order.  After the last one, copy your bluepatcher to the bin directory in your client path and patch it.  This will make the client the correct version for the cluster.


Original post
================================================


For grins i fired up the client to see what I could find.  Status says incompatible protocol.

Also 

7/21/2021 11:52 AM D main: Waiting for incoming connections on port 26000
7/21/2021 11:52 AM T main: Incoming connection
7/21/2021 11:52 AM D Unauthenticated-127.0.0.1: Sending LowLevelVersionExchange...
7/21/2021 11:52 AM E Unauthenticated-127.0.0.1: Exception caught on LowLevelVersionExchange LowLevelVersionExchange must have 6 elements
7/21/2021 11:52 AM E Unauthenticated-127.0.0.1: Exception detected: 
7/21/2021 11:52 AM E Unauthenticated-127.0.0.1: LowLevelVersionExchange must have 6 elements
7/21/2021 11:52 AM T Unauthenticated-127.0.0.1:    at EVE.Packets.LowLevelVersionExchange.op_Implicit(PyDataType data) in X:\EVESharp\EVESharp-master\Server\EVE\Packets\LowLevelVersionExchange.cs:line 24
   at ClusterController.UnauthenticatedConnection.ReceiveLowLevelVersionExchangeCallback(PyDataType ar) in X:\EVESharp\EVESharp-master\Server\ClusterController\UnauthenticatedConnection.cs:line 35
   at Common.Network.EVEClientSocket.FlushReceivingQueue() in X:\EVESharp\EVESharp-master\Server\Common\Network\EVEClientSocket.cs:line 110

**Reply by ixalor** [reply]
Rebuilt the database from scratch doesnt seem like that is the problem as the same error came back up.

**Reply by ixalor** [reply]
I recompiled the source but this time used the debug version.  Same exact error,  Cluster works, node fails, exact same message.
I am not sure if this means there is a problem with my mariadb or something else.  Any help would be awesome.

**Reply by Almamu** [reply]
Always make sure your client version matches the one in the documentation, in this case 6.14.101786 (https://github.com/Almamu/EVESharp/blob/master/Documentation/Pre-requirements.md#eve-online-client). You can see the version on the top left corner. I'll add this to the installation documentation so It's a bit more clear.

**Comment by Almamu** [comment]
About the issue with the ItemManager that you see when starting the Node, that's most likely something missing in the database (either I forgot to add a query on the setup or the setup is not 100% completed). Can you check the following tables? All of them should have this itemID/corporationID in them: 1000132.
```
invItems
eveNames
corporation
crpNPCCorporations
```

I suspect that invItems doesn't have a record of it's own for that item, so the ItemManager doesn't know that an item with that ID exists and doesn't load it on startup. In this case, that ID corresponds to the SCC, which is used by the market services and insurance services for proper creation of orders, insurances, etc...

**Reply by ixalor** [reply]
invitems it shows as the ownerid, but not itemid
eveNames exists.
corporation it is not there.

crpNPCCorporations exists.


I am guessing its the corporation that is missing.

**Reply by ixalor** [reply]
Well, I added the corporation by cloning another one in there but updating the name, ticker and ID and still no dice.  Maybe the invitems is also missing?

**Reply by ixalor** [reply]
Success!

INSERT INTO `invitems` (`itemID`, `typeID`, `ownerID`, `locationID`, `flag`, `contraband`, `singleton`, `quantity`, `customInfo`, `nodeID`) VALUES ('1000132', '0', '0', '0', '0', '0', '1', '1', NULL, '0');

Node fired up, not sure if those are the right settings but with the item at least in the database it did start.

**Reply by ixalor** [reply]
Well that solved one problem and ive found a couple more.

When attempted to create a new player the offices database entry didnt exist.

Added crpoffices.

I did that because I am receiving the error " Cannot find the location of corporation 1000077 (amarr)  1000078 for minmatar 1000015 gallente, 1000044 caldari.

**Reply by ixalor** [reply]
Looks like I need the table crpoffices

I checked and even in the other eve emulators, the crpoffices table does not exist.

**Reply by ixalor** [reply]
Looks like the KEY Command in the crpOffices sql query is not liked:


CREATE TABLE `crpOffices` (
  `corporationID` INT(10) UNSIGNED NOT NULL DEFAULT '0',
  `stationID` INT(10) UNSIGNED NOT NULL DEFAULT '0',
  `officeID` INT(10) UNSIGNED NOT NULL DEFAULT '0',
  `typeID` INT(10) UNSIGNED NOT NULL DEFAULT '0',
  `officeFolderID` INT(10) UNSIGNED NOT NULL DEFAULT '0',
  `startDate` BIGINT(20) NOT NULL DEFAULT '0',
  `rentPeriodInDays` INT(11) NOT NULL DEFAULT '0',
  `periodCost` DOUBLE(22,0) NOT NULL DEFAULT '0',
  `balanceDueDate` DOUBLE(22,0) NULL DEFAULT NULL,
  `nextBillID` INT(10) UNSIGNED NOT NULL,
  PRIMARY KEY  (`corporationID`,`officeFolderID`),
  KEY `itemID` (`itemID`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8;
MySQL said: Documentation

**#1072 - Key column 'itemID' doesn't exist in table**

**Reply by ixalor** [reply]
Yeah after poking around it looks like the corporation table has a total of 8 listed and none of them are the newbie ones that the character creation is asking for.

We have 1000173, 1000174, 1000175,1000176,1000179,180,181,182.

**Reply by ixalor** [reply]
So I removed the key command, ran the script and it worked and added the table however still have the problem of location of corporations not found.

I looked up and added the Amarr default NPC corporations by copying others in the corporation table, giving them the IDs that were missing, changing ticker and name.  Still no go.

**Reply by ixalor** [reply]
Tried adding them to crpstatic, still not working.

7/23/2021 1:29 AM T MachoNet: Calling character::CreateCharacter2
7/23/2021 1:29 AM E character: Cannot find location for corporation 1000165

**Reply by ixalor** [reply]
Interestingly enough, i pulled the crpCorporations data dump and it has much more data, but less columns than yours.

This clear the error, but when creating a character the in-game error  flashes up "failed for some reason" and poofs.

I am guessing due to the database columns which I am going through and poking to see if I can get them lined up with your schema.  Will report back.

**Reply by ixalor** [reply]
Well looks like it created my char.  When I logged back in it was an open.  I tried to load and it failed entering the station.


7/23/2021 4:22 AM T MachoNet: Calling station::GetGuests
7/23/2021 4:22 AM F MachoNet: Detected non-client exception on call to station::GetGuests, registered as error 4. Extra information:
7/23/2021 4:22 AM F MachoNet: Object reference not set to an instance of an object.
7/23/2021 4:22 AM F MachoNet:    at Node.Services.Stations.station.GetGuests(CallInformation call) in X:\2\EVESharp-master\Server\Node\Services\Stations\station.cs:line 46
--- End of stack trace from previous location ---
   at Common.Services.ServiceManager.ServiceCall(String service, String call, PyTuple arguments, Object extraInformation) in X:\2\EVESharp-master\Server\Common\Services\ServiceManager.cs:line 113
   at Node.Network.MachoNet.HandleCallReq(PyPacket packet, Client client) in X:\2\EVESharp-master\Server\Node\Network\MachoNet.cs:line 255
7/23/2021 4:22 AM T MachoNet: Calling warRegistry::MachoResolveObject
7/23/2021 4:22 AM T MachoNet: Calling warRegistry::MachoBindObject
7/23/2021 4:22 AM T BoundService: Calling warRegistry::GetWars on bound service 12

**Reply by Almamu** [reply]
> invitems it shows as the ownerid, but not itemid
> eveNames exists.
> corporation it is not there.
> 
> crpNPCCorporations exists.
> 
> I am guessing its the corporation that is missing.

This is weird, according to Database/Oficial/crpStatic.sql and Database/Server/02-corporation.sql there should be the same data on both tables. Take in mind that crpStatic is not used by the server, but It's a table to have the basic corps, should the need to restore those arise. As it is right now, the tables that are used by live are invItems, eveNames and corporation, which should all have the records for the corporations that are in crpStatic for the server to work properly, and checking all the sql files related to these, they should be there (i see the correct queries to create and import that data between tables...).

This is the query that performs the import of data into invItems, of course the crpStatic table has to be imported first.
```
REPLACE INTO invItems (itemID, typeID, ownerID, locationID, singleton, quantity)
  SELECT crp.corporationID, 2, IF(npc.factionID IS NULL, crp.ceoID, npc.factionID), crp.stationID, 1, 1
    FROM crpStatic AS crp
    LEFT JOIN crpNPCCorporations AS npc USING (corporationID);
```

The error leads me to believe that something went wrong on the database setup. Leaving the crpOffices error aside (which was a real mistake on my side), everything else should be working just fine. If I had to hazard a guess, most likely the Server.sql and Oficial.sql files were not imported or the SQL files were imported in the wrong order. I'd suggest to trash the database and set it up again, following https://github.com/Almamu/EVESharp/blob/master/Documentation/Setup.md closely. Please make sure to import the SQL files in the same order shown in there, as it is important for the progress to work properly. I'd like to simplify it in the near future.

> **#1072 - Key column 'itemID' doesn't exist in table**

I've updated the file to get rid of the error

**Reply by ixalor** [reply]
I have no problem rebuilding the database for the 3rd time.  I am using mariadb on windows 10, and importing the sql using heidiSQ.  The first file alone takes 3hrs to add to the database.  I have backed up the database, and I am trying commandline mysql to source the files instead of importing them via HeidiSQL.  I will let you know my results.

**Reply by Almamu** [reply]
> I have no problem rebuilding the database for the 3rd time. I am using mariadb on windows 10, and importing the sql using heidiSQ. The first file alone takes 3hrs to add to the database. I have backed up the database, and I am trying commandline mysql to source the files instead of importing them via HeidiSQL. I will let you know my results.

Importing the files through the command line should be way faster than HeidiSQL, usually takes no more than 15 minutes on my side. I have not done any setup with HeidiSQL, so I'm not really sure how well It'll manage to do the setup...

**Reply by ixalor** [reply]
wow you are right, this is so much faster. First file already done!  Ill fire up the server to see if it works and report back.

**Reply by ixalor** [reply]
Well same error as before, the import was much faster commandline but still missing the default corporation data.

7/23/2021 5:04 AM T MachoNet: Calling character::GetAppearanceInfo
7/23/2021 5:04 AM T MachoNet: Calling character::ValidateNameEx
7/23/2021 5:05 AM T MachoNet: Calling character::ValidateNameEx
7/23/2021 5:05 AM T MachoNet: Calling character::ValidateNameEx
7/23/2021 5:05 AM T MachoNet: Calling character::CreateCharacter2
7/23/2021 5:05 AM E character: Cannot find location for corporation 1000166

**Reply by ixalor** [reply]
Also hyphens should work in the char surname and it is blocked :) (minor)

**Reply by ixalor** [reply]
SQL errors:

ERROR 1101 (42000) at line 900 in file: 'Server.sql': BLOB/TEXT column 'description' can't have a default value
Query OK, 0 rows affected (0.00 sec)

ERROR 1101 (42000) at line 944 in file: 'Server.sql': BLOB/TEXT column 'comment' can't have a default value
Query OK, 0 rows affected (0.00 sec)

ERROR 1101 (42000) at line 961 in file: 'Server.sql': BLOB/TEXT column 'body' can't have a default value
Query OK, 0 rows affected (0.00 sec)
ERROR 1072 (42000) at line 822 in file: 'Server.sql': Key column 'itemID' doesn't exist in table
Query OK, 0 rows affected (0.00 sec)

ERROR 1064 (42000) at line 1325 in file: 'Server.sql': You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near 'UNION (SELECT 0) LIMIT 1)' at line 1
Query OK, 2 rows affected (0.00 sec)
Records: 2  Duplicates: 0  Warnings: 0

**Reply by ixalor** [reply]
Deleted database, rerunning with sql logging going so I can get all errors to share.

**Reply by Almamu** [reply]
Please include all the sql commands you ran in the process.

**Reply by ixalor** [reply]
mysqld -u root -p 
password
tee sql.log

Then literally followed the steps in your howto.  Just copy and paste (minus changing database password)

CREATE DATABASE evedb;
drop user evesharp@localhost (wont let you create, it thinks is still there)

CREATE USER 'evesharp'@'localhost' IDENTIFIED BY 'password1';
GRANT ALL PRIVILEGES ON evedb.* TO 'evesharp'@'localhost';


source apo15-mysql5-v1.sql;
source mapPrecalculatedSolarSystemJumps.sql;
source Oficial.sql;
source Static.sql;
source Dynamic.sql;
source Server.sql;

I wait for each to stop then run the next one once im back at a mysql prompt. Im in the middle of apo15 atm.

edit: between files now on maps i am changing the log to sql2,3 etc so each file has its own error log.

**Reply by ixalor** [reply]
Okay so no errors in the first two files.  Oficial.sql:

ERROR 1146 (42S02): Table 'evedb.crsstatic' doesn't exist
Query OK, 0 rows affected, 1 warning (0.00 sec)


No errors in static.
no errors in dynamic

Error in server:




ERROR 1054 (42S22): Unknown column 'typeID' in 'field list'
Query OK, 8 rows affected (0.00 sec)

ERROR 1064 (42000): You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near '
  CONSTRAINT `fk_characterID`
    FOREIGN KEY (`characterID`)
    REFERENCES `c' at line 6


ERROR 1072 (42000): Key column 'itemID' doesn't exist in table
Query OK, 0 rows affected, 1 warning (0.00 sec)

ERROR 1101 (42000): BLOB/TEXT column 'description' can't have a default value
Query OK, 0 rows affected, 1 warning (0.00 sec)


ERROR 1101 (42000): BLOB/TEXT column 'comment' can't have a default value
Query OK, 0 rows affected, 1 warning (0.00 sec)

ERROR 1101 (42000): BLOB/TEXT column 'body' can't have a default value
Query OK, 0 rows affected, 1 warning (0.00 sec)


ERROR 1064 (42000): You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near 'UNION (SELECT 0) LIMIT 1)' at line 1
Query OK, 2 rows affected (0.00 sec)

**Reply by ixalor** [reply]
first error corrected by fixing a typo in the input from crsstatic to crpstatic and reran it.  Now crpstatic has all the corporations listed.

Line 239 updated to crpstatic.




  Checking the other errors.

**Reply by ixalor** [reply]
Deleted everything and downloaded and rebuilt the database after the changes you made.   Still a few errors.  The node does run now, i can make the character, but it fails on trying to launch.  I reload the game and see the character there, use it and I load the dock however i cannot leave the dock it errors.



7/24/2021 6:51 AM F MachoNet: Detected non-client exception on call to ship::MachoBindObject, registered as error 5. Extra information:
7/24/2021 6:51 AM F MachoNet: Cannot find an appropiate function definition for Undock on service (boundID 14) ship
7/24/2021 6:51 AM F MachoNet:    at Common.Services.ServiceManager.FindMethods(IService service, String serviceName, String method) in X:\EVESharp\EVESharp-master\Server\Common\Services\ServiceManager.cs:line 34
   at Node.Network.BoundServiceManager.ServiceCall(Int32 boundID, String call, PyTuple payload, CallInformation callInformation) in X:\EVESharp\EVESharp-master\Server\Node\Network\BoundServiceManager.cs:line 112
   at Node.Network.ClientBoundService.MachoBindObject(ServiceBindParams bindParams, PyDataType callInfo, CallInformation call) in X:\EVESharp\EVESharp-master\Server\Node\Network\ClientBoundService.cs:line 84
   at Node.Network.BoundService.MachoBindObject(PyDataType bindParams, PyDataType callInfo, CallInformation call) in X:\EVESharp\EVESharp-master\Server\Node\Network\BoundService.cs:line 78
--- End of stack trace from previous location ---
   at Common.Services.ServiceManager.ServiceCall(String service, String call, PyTuple arguments, Object extraInformation) in X:\EVESharp\EVESharp-master\Server\Common\Services\ServiceManager.cs:line 116
   at Node.Network.MachoNet.HandleCallReq(PyPacket packet, Client client) in X:\EVESharp\EVESharp-master\Server\Node\Network\MachoNet.cs:line 293

While I have no issues continuing to troubleshoot the database install/setup I do have to ask the obvious question: If you have a completely working database that allows the game to play why not just dump it and post it?

**Reply by ixalor** [reply]
Going down the list of every file in the server directory

04-entity:

ERROR 1054 (42S22) at line 136 in file: '04-entity.sql': Unknown column 'typeID' in 'field list'

 * Insert static characters to invItems table
 */
REPLACE INTO invItems (itemID, typeID, ownerID, locationID, singleton, quantity)
 SELECT characterID, typeID, 1, stationID, 1, 1
  FROM chrInformation WHERE characterID < 100000000;






MariaDB [evedb]> source 05-chrFriends.sql
Query OK, 0 rows affected, 1 warning (0.00 sec)

ERROR 1064 (42000) at line 3 in file: '05-chrFriends.sql': You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near '
  CONSTRAINT `fk_characterID`
    FOREIGN KEY (`characterID`)
    REFERENCES `c' at line 6


ERROR 1101 (42000) at line 3 in file: '21-crprecruitmentads.sql': BLOB/TEXT column 'description' can't have a default value

ERROR 1101 (42000) at line 3 in file: '24-chrbookmarks.sql': BLOB/TEXT column 'comment' can't have a default value

ERROR 1101 (42000) at line 5 in file: '25-evemail.sql': BLOB/TEXT column 'body' can't have a default value

MariaDB [evedb]> source 43-crpceos.sql
ERROR 1064 (42000) at line 5 in file: '43-crpceos.sql': You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near 'UNION (SELECT 0) LIMIT 1)' at line 1

I fixed the easy ones with no default allowed by removing the default part and rerunning.

What i dont get is why any of these would prevent me from undocking :(

**Reply by ixalor** [reply]
Node window:

7/24/2021 9:11 AM T BoundService: Calling ship::Undock on bound service 18
7/24/2021 9:11 AM F MachoNet: Detected non-client exception on call to ship::MachoBindObject, registered as error 3. Extra information:
7/24/2021 9:11 AM F MachoNet: Cannot find an appropiate function definition for Undock on service (boundID 18) ship
7/24/2021 9:11 AM F MachoNet:    at Common.Services.ServiceManager.FindMethods(IService service, String serviceName, String method) in X:\2\EVESharp-master\Server\Common\Services\ServiceManager.cs:line 34
   at Node.Network.BoundServiceManager.ServiceCall(Int32 boundID, String call, PyTuple payload, CallInformation callInformation) in X:\2\EVESharp-master\Server\Node\Network\BoundServiceManager.cs:line 112
   at Node.Network.ClientBoundService.MachoBindObject(ServiceBindParams bindParams, PyDataType callInfo, CallInformation call) in X:\2\EVESharp-master\Server\Node\Network\ClientBoundService.cs:line 84
   at Node.Network.BoundService.MachoBindObject(PyDataType bindParams, PyDataType callInfo, CallInformation call) in X:\2\EVESharp-master\Server\Node\Network\BoundService.cs:line 78
--- End of stack trace from previous location ---
   at Common.Services.ServiceManager.ServiceCall(String service, String call, PyTuple arguments, Object extraInformation) in X:\2\EVESharp-master\Server\Common\Services\ServiceManager.cs:line 116
   at Node.Network.MachoNet.HandleCallReq(PyPacket packet, Client client) in X:\2\EVESharp-master\Server\Node\Network\MachoNet.cs:line 293
7/24/2021 9:11 AM T MachoNet: Calling station::GetGuests

**Reply by Almamu** [reply]
> Deleted everything and downloaded and rebuilt the database after the changes you made. Still a few errors. The node does run now, i can make the character, but it fails on trying to launch. I reload the game and see the character there, use it and I load the dock however i cannot leave the dock it errors.
> 
> 7/24/2021 6:51 AM F MachoNet: Detected non-client exception on call to ship::MachoBindObject, registered as error 5. Extra information:
> 7/24/2021 6:51 AM F MachoNet: Cannot find an appropiate function definition for Undock on service (boundID 14) ship
> 7/24/2021 6:51 AM F MachoNet: at Common.Services.ServiceManager.FindMethods(IService service, String serviceName, String method) in X:\EVESharp\EVESharp-master\Server\Common\Services\ServiceManager.cs:line 34
> at Node.Network.BoundServiceManager.ServiceCall(Int32 boundID, String call, PyTuple payload, CallInformation callInformation) in X:\EVESharp\EVESharp-master\Server\Node\Network\BoundServiceManager.cs:line 112
> at Node.Network.ClientBoundService.MachoBindObject(ServiceBindParams bindParams, PyDataType callInfo, CallInformation call) in X:\EVESharp\EVESharp-master\Server\Node\Network\ClientBoundService.cs:line 84
> at Node.Network.BoundService.MachoBindObject(PyDataType bindParams, PyDataType callInfo, CallInformation call) in X:\EVESharp\EVESharp-master\Server\Node\Network\BoundService.cs:line 78
> --- End of stack trace from previous location ---
> at Common.Services.ServiceManager.ServiceCall(String service, String call, PyTuple arguments, Object extraInformation) in X:\EVESharp\EVESharp-master\Server\Common\Services\ServiceManager.cs:line 116
> at Node.Network.MachoNet.HandleCallReq(PyPacket packet, Client client) in X:\EVESharp\EVESharp-master\Server\Node\Network\MachoNet.cs:line 293
> 
> While I have no issues continuing to troubleshoot the database install/setup I do have to ask the obvious question: If you have a completely working database that allows the game to play why not just dump it and post it?

Undocking is not supported yet, most of the supported stuff is just in-station things, like inventory, corps, chat, mail... I'll take a look at the database issues you've had with the invItems table creation and patch those issues today.

About the database dump, the main issue is that my database copy has a lot of crap that I've left there while doing testing and development. I usually make a new database installation from time to time, so It takes me some time to notice these kind of installation issues. I'll try to do one today and get all the issues fixed, but most of the errors you see should not prevent normal operation of most of the supported features (I guess friends list won't work and the info window for some corps won't display properly, maybe evemail won't work if the table was not created). By the way what MariaDB version are you using for this installation? The "UNION (SELECT 0) LIMIT 1) should work as that query was copied directly from the execute window of HeidiSQL after running it...

**Reply by ixalor** [reply]
mysql  Ver 15.1 Distrib 10.0.21-MariaDB, for Win64 (AMD64)

I started using commandline only btw.

**Reply by Almamu** [reply]
I've done some testing on my side. With MariaDB 10.5 none of the errors you shared happen, except for one, which is a missing field in a table that I've already fixed and uploaded the change for. I've also updated the documentation to reflect MariaDB 10.5 as a minimum requirement instead of "any" MariaDB version, and from now on, only 10.5 will be supported, although this might change in the future. Please pull the latest changes and rebuild the database. It should work fine now. Remember that undocking is not supported yet.
